streaming/src/test/scala/org/apache/spark/streaming/InputStreamsSuite.scala

```diff
@@ -235,18 +236,97 @@ class InputStreamsSuite extends TestSuiteBase with BeforeAndAfter {
         }
         // Over time, create files in the temp directory 1
         val input1 = Seq(1, 2, 3, 4, 5)
-        input1.foreach(i => createFileAndAdvenceTime(i, testSubDir1))
+        input1.foreach(i => createFileAndAdvanceTime(i, testSubDir1))
 
         // Over time, create files in the temp directory 1
         val input2 = Seq(6, 7, 8, 9, 10)
-        input2.foreach(i => createFileAndAdvenceTime(i, testSubDir2))
+        input2.foreach(i => createFileAndAdvanceTime(i, testSubDir2))
 
         // Verify that all the files have been read
         val expectedOutput = (input1 ++ input2).map(_.toString).toSet
         assert(outputQueue.asScala.flatten.toSet === expectedOutput)
       }
     } finally {
-      if (testDir != null) Utils.deleteRecursively(testDir)
+      if (testDir != null) {
+        Utils.deleteRecursively(testDir)
+      }
+    }
+  }
+
+  /**
+   * Tests that renamed directories are included in new batches -but that only files created
+   * within the batch window are included.
+   * Uses the Hadoop APIs to verify consistent behavior with the operations used internally.
+   */
+  test("renamed directories are scanned") {
+    val testDir = Utils.createTempDir()
```

Comment:
Can you use `withTempDir { testDir =>`? if so I think several tests could use the same pattern here.